src/v0/destinations/klaviyo/transform.js

```diff
@@ -140,55 +171,63 @@ const trackRequestHandler = (message, category, destination) => {
           }
         });
       }
-      if (!payload.properties) {
+      if (!attributes.properties) {
         payload.properties = {};
       }
       if (itemArr.length > 0) {
-        payload.properties.items = itemArr;
+        attributes.properties.items = itemArr;
       }
     }
 
     // all extra props passed is incorporated inside properties
     let customProperties = {};
     customProperties = extractCustomFields(
       message,
-      customProperties,
+      attributes.customProperties,
       ['properties'],
       ecomExclusionKeys,
     );
     if (!isEmptyObject(customProperties)) {
-      payload.properties = {
-        ...payload.properties,
+      attributes.properties = {
+        ...attributes.properties,
         ...customProperties,
       };
     }
 
-    if (isEmptyObject(payload.properties)) {
+    if (isEmptyObject(attributes.properties)) {
       delete payload.properties;
     }
   } else {
-    payload = constructPayload(message, MAPPING_CONFIG[category.name]);
-    payload.token = destination.Config.publicApiKey;
+    attributes = constructPayload(message, MAPPING_CONFIG[category.name]);
+    // payload.token = destination.Config.publicApiKey;
     if (message.properties && message.properties.revenue) {
-      payload.properties.$value = message.properties.revenue;
-      delete payload.properties.revenue;
+      attributes.value = message.properties.revenue;
+      delete attributes.properties.revenue;
     }
     const customerProperties = createCustomerProperties(message);
-    if (destination.Config.enforceEmailAsPrimary) {
-      delete customerProperties.$id;
-      customerProperties._id = getFieldValueFromMessage(message, 'userId');
-    }
-    payload.customer_properties = customerProperties;
+    attributes.properties = {
+      ...attributes.properties,
+      ...populateCustomFieldsFromTraits(message),
+    };
+    // if (destination.Config.enforceEmailAsPrimary) {
+    //   delete customerProperties.external_id;
+    //   customerProperties._id = getFieldValueFromMessage(message, 'userId');
+    // }
+    attributes.profile = customerProperties;
```

Comment:
```suggestion
    attributes.profile = createCustomerProperties(message);
```
No need to create a variable for that

Reply:
done